### ISSUE TITLE
Mix the STT engines

### ISSUE THREAD

**Issue by Fortaweb** [alexylem/jarvis#223]
Hi,

I was wondering if it is possible for you to make it customizable which STT engine you use for a command. Because I really like snowboy for it's accuracy, but I still need wit.ai for 'user input' like the title of a song. I tried to find how to add it myself but I'm not familiar enough with this.

I think it should be like you have the default stt engine, but with a function in your command you can make the next input be processed with a different engine.

I also found a few bugs, at least which happened to me:
- If jarvis is running and you open ./jarvis.sh I can't get back to the process, and I also can't kill it.
- Sometimes when I edit a setting it doesn't save. (I think wit access token was one of them).

By the way, thanks for me letting me know here (https://www.raspberrypi.org/forums/viewtopic.php?f=31&t=150031&p=986075) that you improved this script a lot. The voice recognition really improved (not only due to snowboy but also it captures my mic a little bit better).

Best regards,
Menno

**Comment by alexylem** [comment]
Hello @Fortaweb 
Actually Jarvis has been built around the idea to use 2 SST engines, one for the hotword and one for the commands.
This is the way I recommend to use it, so in your settings you have to set snowboy for the trigger and bing/wit for the commands. (Bing is a little better).
On top of this, you can also train snowboy to recognize full commands (that are 100% predictive, not containing any variable text).
Please let me know if that is ok for you.

For the issues, could you please try to reproduce? the finding of the process normally works (except if Jarvis crashes for some reason). For the settings they are saved when you go back to the main menu, including wit token. Please do retry.

Alex.

**Comment by alexylem** [comment]
@Fortaweb any news?

**Comment by Fortaweb** [comment]
Hi,

Maybe I didn't look good enough but where can I add extra snowboy commands?

And I got a little further with those issues:
- it might be because I push control + z when I want to edit the config after starting Jarvis, but after I made some changes it can't find the log or something (I don't remember the full error).
- that it doesn't save makes sense because as I said before I have the bad habbit of using control + z alot for quiting something.

Menno
